repos/fdbt-site/src/pages/serviceList.tsx

```diff
@@ -161,12 +166,13 @@ export const getServerSideProps = async (ctx: NextPageContextWithSession): Promi
     const nocCode = getAndValidateNoc(ctx);
     const serviceListAttribute = getSessionAttribute(ctx.req, SERVICE_LIST_ATTRIBUTE);
     let dataSourceAttribute = getSessionAttribute(ctx.req, TXC_SOURCE_ATTRIBUTE);
+    const modesAttribute = getSessionAttribute(ctx.req, MULTI_MODAL_ATTRIBUTE);
 
     if (!dataSourceAttribute) {
         const services = await getAllServicesByNocCode(nocCode);
         const hasBodsServices = services.some((service) => service.dataSource && service.dataSource === 'bods');
 
-        if (!hasBodsServices) {
+        if (!hasBodsServices && !modesAttribute) {
```

Comment:
What happens if the user logs in and they have no BODS data, and they also have no modes? Wont they have a modesAttribute with an empty array?

Reply:
no, we are setting the attribute only when they have modes in it